lib/python-caller-trampoline.py

```diff
@@ -56,6 +57,9 @@
         if hasattr(mod, fcn):
             # call the desired function in the loaded module
             method = getattr(mod, fcn)
+            for arg in args:
+                if isinstance(arg, dict):
+                    decode_from_json(arg)
```

Comment:
Should we white-list the keys of `data` that this operates on? In particular, `raw_submitted_input` is the only thing directly under user control, and that's a major risk to process. We could just black-list this one key?

Reply:
Sure, I should have done that already. Let me know what you prefer, a white-list with `params`, `correct_answers`, `submitted_answers` or a black-list with `raw_submitted_answers`.

Reply:
Let's whitelist